### ISSUE TITLE
Don't require IPC_LOCK and SYS_RESOURCE capabilities by default

### ISSUE THREAD

**Issue by johananl** [instaclustr/cassandra-operator#237]
We should consider removing the extra capabilities which are currently added to the Cassandra containers by the operator:

https://github.com/instaclustr/cassandra-operator/blob/561616649cd86cf8e59e25f03baa55000892fc25/pkg/controller/cassandradatacenter/statefulset.go#L119-L126

Following https://github.com/instaclustr/cassandra-operator/pull/219#discussion_r310695441 it is likely that we don't actually need these capabilities in the Cassandra container since things seem to work without them (albeit with potentially reduced performance). We may want to conditionally add these capabilities only when the sysctl support flag discussed in https://github.com/instaclustr/cassandra-operator/issues/208 is set to `true`.

**Comment by smiklosovic** [comment]
Seems as a good idea to me. What is the progress on 208?

**Comment by johananl** [comment]
@smiklosovic looks like we are [waiting](https://github.com/instaclustr/cassandra-operator/issues/208#issuecomment-518497328) for a response from @zegelin and @benbromhead.

I can take the initiative there and propose a solution for #208 following my [comment](https://github.com/instaclustr/cassandra-operator/issues/208#issuecomment-515934889) in a new PR, but I thought @alourie was on it so didn't want to interfere.

**Comment by alourie** [comment]
@johananl @benbromhead @zegelin yes, I am basically waiting on the comment in #208. I can do this either stand alone or as a part of #218.

**Comment by DestyNova** [comment]
Hello, is there a workaround to avoid this problem? As it stands, I can't deploy Cassandra in one of our test clusters:

```
Warning  FailedCreate      60s (x26 over 6m16s)  statefulset-controller  create
  Pod cassandra-eolas-dc-0 in StatefulSet cassandra-eolas-dc failed error: pods
  "cassandra-eolas-dc-0" is forbidden: unable to validate against any pod security
  policy: [spec.initContainers[0].securityContext.privileged: Invalid value: true:
  Privileged containers are not allowed spec.containers[0].securityContext.capabilities.add:
  Invalid value: "IPC_LOCK": capability may not be added
  spec.containers[0].securityContext.capabilities.add:
  Invalid value: "SYS_RESOURCE": capability may not be added]
```

**Comment by johananl** [comment]
Hello @DestyNova. Could you provide some information regarding your setup?

- The version of the operator you are running.
- Manifest used to create the Cassandra instance.

Relevant work is being done in https://github.com/instaclustr/cassandra-operator/issues/208.

As a workaround in the meantime, you may be able to tweak the relevant pod security policy in your cluster to allow the following:

- Privileged containers
- The `IPC_LOCK` capability
- The `SYS_RESOURCE` capability

If you are not sure which PSP to edit, the following may give you a hint:

```
kubectl get pods <cassandra pod> -oyaml | grep psp
```

**Comment by DestyNova** [comment]
Thanks @johananl! I don't have access now, but I'll try that tomorrow :+1:

**Comment by DestyNova** [comment]
@johananl I'm not sure what the version is, but::
```
image: gcr.io/cassandra-operator/cassandra-operator:latest
    imageID: docker-pullable://gcr.io/cassandra-operator/cassandra-operator@sha256:bf3a2d21f4036a9bc267c66bb4ed3fd6da0a8d94ec6db2b021ce6bb3bb3474b4
```

**Comment by johananl** [comment]
Fixed in https://github.com/instaclustr/cassandra-operator/pull/269.
